langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/transport/http/HttpMcpTransport.java

```diff
@@ -0,0 +1,177 @@
+package dev.langchain4j.mcp.client.transport.http;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.langchain4j.mcp.client.protocol.McpCallToolRequest;
+import dev.langchain4j.mcp.client.protocol.McpInitializeRequest;
+import dev.langchain4j.mcp.client.protocol.McpListToolsRequest;
+import dev.langchain4j.mcp.client.transport.McpTransport;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.sse.EventSource;
+import okhttp3.sse.EventSources;
+
+public class HttpMcpTransport implements McpTransport {
```

Comment:
How would this be overridden in Quarkus?

Reply:
Quarkus will have its own implementation of `McpTransport` that doesn't depend  on OkHttp. The stdio  transport should work as-is, I assume

Reply:
👍🏽

Reply:
I tried to only include the okhttp-specific stuff here as much as I could, most reusable stuff is split out into other places